### PR TITLE
Rope Bug Fix + Underground transition

### PR DESCRIPTION
**Pull request by zahranikbakht**
This PR should include the following:
- Rope Bug Fix
- Transition to underground (animation + fade in/out)
- Ladder always faces the camera

### CODE REVIEW

**Review by danielwiktorczyk — CHANGES_REQUESTED** [review]
I think Elizabeth mentioned this, but the player isn't flush to the underground

**Review by tiffhc — APPROVED** [review]
Should be good. Just check the logic of the code with the other programmers before merging back!

File: Assets/Scripts/Digging/DiggingAndRopeInteractions.cs [hunk 1]
```diff
@@ -151,6 +155,9 @@ public IEnumerator ThrowRope(GameObject ropeObject, Tile tile)
         playerMovement.MoveTowards(Vector3.zero, 0);
         ropeObject.transform.position = new Vector3(tile.transform.position.x, ropeObject.transform.position.y, tile.transform.position.z);
         ropeObject.SetActive(true);
+        rope.GetUpperLadder().transform.forward = Vector3.back;
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Digging/DiggingAndRopeInteractions.cs:158]: Why not Vector3.forward?

File: Assets/Scripts/Player/PlayerMovement.cs [hunk 2]
```diff
@@ -69,7 +66,7 @@ void PlayerMovementUpdate()
 
         CheckIfFalling();
 
-        if (this.isFalling)
+        if (this.isFalling || this.isInFallingProcess)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Player/PlayerMovement.cs:69]: These two booleans mean the same thing... maybe there's a way to combine the two? Or differentiate between them?

File: Assets/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -101,11 +95,8 @@ private void RegularMovement()
 
     private void FallingMovementUpdate()
     {
-        if (!this.fallingCamera.activeSelf)
-            this.fallingCamera.SetActive(true);
-
-        var fallingVelocity = Vector3.down * Time.deltaTime * this.fallingSpeed;
-        this.characterController.Move(fallingVelocity);
+        if (!isInFallingProcess)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Player/PlayerMovement.cs:98]: Yah, I think there's a way of avoiding this boolean altogether

File: Assets/Scripts/Player/PlayerMovement.cs [hunk 4]
```diff
@@ -210,4 +201,18 @@ public IEnumerator FadeOut()
...
+    {
+        isInFallingProcess = true;
+
+        yield return StartCoroutine(FadeIn());
+        this.transform.position +=  Vector3.up * (this.undergroundCheckThreshold + this.underground.transform.position.y - transform.position.y - 1f);
+        this.virtualCamera.SetActive(false);
+        this.virtualCamera.SetActive(true);
+        yield return StartCoroutine(FadeOut());
+        
+        isInFallingProcess = false;
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Player/PlayerMovement.cs:214]: if you don't use this `isInFallingProcess` between this method, then there's no point of switching it from true, then to false. I think you can avoid this bool altogether

**Review by danielwiktorczyk — CHANGES_REQUESTED** [review]
Check that extra boolean out! Otherwise, looking good!

File: Assets/Scripts/Digging/DiggingAndRopeInteractions.cs (hunk 1 above)
Comment (zahranikbakht) [reply Assets/Scripts/Digging/DiggingAndRopeInteractions.cs:158]: the rope model has it backwards. So if I set it to Vector3.forward then the rope is reversed .

File: Assets/Scripts/Player/PlayerMovement.cs (hunk 2 above)
Comment (zahranikbakht) [reply Assets/Scripts/Player/PlayerMovement.cs:69]: I had to create this boolean for the coroutine. Note that the coroutine is started in the Update method. To make sure that the coroutine is started only once and not every frame, I am using this boolean.

**Review by danielwiktorczyk — APPROVED** [review]
Worked on it with you. Woo!

**Pull request merged**